### ISSUE TITLE
I demand you to make Windows support 

### ISSUE THREAD

**Issue by AndreyStille** [NVIDIA/nccl#1122]
For god's sake you are selling plenty of overpriced GPUs constantly. Why can't you spend one week of teamwork to make your soft suitable for every OS? 
Rather than doing some more shitty ML apps

**Comment by sjeaugey** [comment]
Good news is, we're actually working on this. It's not "one week of work" though. People may not realize it but it's not just about porting the code to Windows. That itself will take a lot more than a week. But it's also about maintaining that support in the long term, i.e. making sure it still works on every release, checking for performance regressions, fixing bugs when users report them, etc.
In other words, we need to build a full infrastructure which runs on Windows.
Even for the code base, it has a significant long term cost, both to maintain and as it makes any new code more expensive to develop, and makes the barrier higher for anyone to contribute to the project.

**Comment by MonroeD** [comment]
> Good news is, we're actually working on this. It's not "one week of work" though. People may not realize it but it's not just about porting the code to Windows. That itself will take a lot more than a week. But it's also about maintaining that support in the long term, i.e. making sure it still works on every release, checking for performance regressions, fixing bugs when users report them, etc. In other words, we need to build a full infrastructure which runs on Windows. Even for the code base, it has a significant long term cost, both to maintain and as it makes any new code more expensive to develop, and makes the barrier higher for anyone to contribute to the project.

does cuda support cudaIpc in windows?  if not, how support multi process in nccl on windows.
